subprojects/kotlin-dsl-test-fixtures/src/main/kotlin/org/gradle/kotlin/dsl/fixtures/zipUtils.kt

```diff
@@ -1,14 +1,37 @@
 package org.gradle.kotlin.dsl.fixtures
 
+import org.gradle.kotlin.dsl.support.zipTo
+
+import java.io.File
+
 import kotlin.reflect.KClass
 
 
-fun classEntriesFor(classes: Array<out KClass<*>>) =
-    classEntriesFor(*classes.map { it.java }.toTypedArray())
+fun classEntriesFor(classes: Array<out KClass<*>>): Sequence<Pair<String, ByteArray>> =
+    classes.asSequence().map { classEntryFor(it) }
 
 
 fun classEntriesFor(vararg classes: Class<*>): Sequence<Pair<String, ByteArray>> =
-    classes.asSequence().map {
-        val classFilePath = it.name.replace('.', '/') + ".class"
-        classFilePath to it.getResource("/$classFilePath").readBytes()
+    classes.asSequence().map { classEntryFor(it) }
+
+
+fun classEntryFor(clazz: KClass<*>): Pair<String, ByteArray> =
+    classEntryFor(clazz.java)
+
+
+fun classEntryFor(clazz: Class<*>): Pair<String, ByteArray> {
+    val classFilePath = clazz.name.replace('.', '/') + ".class"
+    return classFilePath to clazz.getResource("/$classFilePath").readBytes()
+}
+
+
+fun jarWithPluginDescriptors(file: File, vararg pluginIdsToImplClasses: Pair<String, String>) =
```

Comment:
❓  what about moving this function and the one below to a less generic place, like in a sibling `plugins.kt` fixture file or something?

Reply:
Done in 68e4c52927e09738f7be11cf4ddf3a06e08b6150.